client/signup/jetpack-connect/authorize-form.jsx

```diff
@@ -640,24 +636,27 @@ const JetpackConnectAuthorizeForm = React.createClass( {
 
 export default connect(
 	state => {
-		const site = state.jetpackConnect.jetpackConnectAuthorize &&
-				state.jetpackConnect.jetpackConnectAuthorize.queryObject &&
-				state.jetpackConnect.jetpackConnectAuthorize.queryObject.site
+		const queryObjectSite = state.jetpackConnect.jetpackConnectAuthorize &&
+			state.jetpackConnect.jetpackConnectAuthorize.queryObject &&
+			state.jetpackConnect.jetpackConnectAuthorize.queryObject.site;
+
+		const site = queryObjectSite
 			? getSiteByUrl( state, state.jetpackConnect.jetpackConnectAuthorize.queryObject.site )
 			: null;
 		const requestHasXmlrpcError = () => {
-			return hasXmlrpcError( state.jetpackConnect.jetpackConnectAuthorize );
+			return hasXmlrpcError( state );
 		};
 		const isFetchingSites = () => {
 			return isRequestingSites( state );
 		};
+
 		return {
 			jetpackConnectAuthorize: state.jetpackConnect.jetpackConnectAuthorize,
 			jetpackSSOSessions: state.jetpackConnect.jetpackSSOSessions,
-			jetpackConnectSessions: state.jetpackConnect.jetpackConnectSessions,
 			isAlreadyOnSitesList: !! site,
 			isFetchingSites,
-			requestHasXmlrpcError
+			requestHasXmlrpcError,
+			calypsoStartedConnection: isCalypsoStartedConnection( state, queryObjectSite )
```

Comment:
Hmm, we can't do that, because we are also using this selector in other parts of the code that passes a different parameter (also a site slug, but obtained from a different subtree).  Maybe we should make this an abstract selector and create a couple of new ones that deal with each case, but I'm not sure about if it would be adding more complexity for the sake of purity